pkg/controllers/federate/controller.go

```diff
@@ -97,13 +99,14 @@ func NewFederateController(
 	fedSystemNamespace string,
 ) (*FederateController, error) {
 	controllerName := fmt.Sprintf("%s-federate-controller", typeConfig.GetFederatedType().Name)
+	logger := klog.LoggerWithValues(klog.Background(), "controller", FederateControllerName, "ftc", typeConfig.Name)
```

Comment:
Should we pass a parent logger into the constructor to allow top-level configuraitons?

Reply:
what top level configurations do we need? i think this is ok for now, in the future we may retrieve a parent logger from the context passed to the reconcile function (once the worker is refactored)

Reply:
Configurations like log sink, level, etc.